### PR TITLE
[posix] add missing header guards, style fixes

### PR DESCRIPTION
**Pull request by abtink**
This commit contains smaller changes in the `posix` platform modules:
- Adds missing header guard checks.
- Use consistent names for header guard defines.
- Fix/update comments.

### CODE REVIEW

**Review by bukepo — APPROVED** [review]

File: src/posix/platform/power.hpp [hunk 1]
```diff
@@ -26,8 +26,8 @@
  *  POSSIBILITY OF SUCH DAMAGE.
  */
 
-#ifndef POSIX_PLATFORM_POWER_H
-#define POSIX_PLATFORM_POWER_H
+#ifndef OT_POSIX_PLATFORM_POWER_H
+#define OT_POSIX_PLATFORM_POWER_H
```
Comment (superwhd) [inline src/posix/platform/power.hpp:30]: Should it ends with `HPP_`?

File: src/posix/platform/resolver.hpp [hunk 2]
```diff
@@ -73,19 +73,15 @@ class Resolver
     /**
      * Updates the file descriptor sets with file descriptors used by the radio driver.
      *
-     * @param[in,out]  aReadFdSet   A reference to the read file descriptors.
-     * @param[in,out]  aErrorFdSet  A reference to the error file descriptors.
-     * @param[in,out]  aMaxFd       A reference to the max file descriptor.
-     * @param[in,out]  aTimeout     A reference to the timeout.
+     * @param[in,out]  aContext   The mainloop context.
```
Comment (superwhd) [inline src/posix/platform/resolver.hpp:76]:
```suggestion
     * @param[in,out]  aContext  The mainloop context.
```
nit

File: src/posix/platform/power.hpp (hunk 1 above)
Comment (abtink) [reply src/posix/platform/power.hpp]: Yes. It should. Thanks.

**Review by superwhd — APPROVED** [review]
LGTM, thanks! 👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**